dev/io.openliberty.security.oidcclientcore.internal/src/io/openliberty/security/oidcclientcore/utils/CommonJose4jUtils.java

```diff
@@ -61,5 +71,140 @@ public JsonWebStructure getJsonWebStructureFromJwtContext(JwtContext jwtContext)
         }
         return jsonStruct;
     }
+    
+    public TokenSignatureValidationBuilder signaturevalidationbuilder() {
+        return this.tokenSignatureValidationBuilder;
+    }
+    
+  //TODO: use this in regular OIDC flow to do token signature validation
+
+    public static class TokenSignatureValidationBuilder {
+
+        private JsonWebStructure signature;
+        private SSLSupport sslsupport;
+        private String jwkuri;
+        private String clientid;
+        private String clientsecret;
+        JWKSet jwkset = new JWKSet(); // TODO: this should be per Client?
+        private String issuer;
+
+        private TokenSignatureValidationBuilder() {
+
+        }
+
+        /**
+         * @param jsonStruct
+         */
+        public TokenSignatureValidationBuilder signature(JsonWebStructure jsonStruct) {
+            this.signature = jsonStruct;
+            return this;
+        }
+
+        /**
+         * @param sslSupport
+         */
+        public TokenSignatureValidationBuilder sslsupport(SSLSupport sslSupport) {
+            this.sslsupport = sslSupport;
+            return this;
+        }
+
+        /**
+         * @param jwksURI
+         */
+        public TokenSignatureValidationBuilder jwkuri(String jwksURI) {
+            this.jwkuri = jwksURI;
+            return this;
+        }
+
+        /**
+         * @param clientId
+         */
+        public TokenSignatureValidationBuilder clientid(String clientId) {
+            this.clientid = clientId;
+            return this;
+        }
+
+        /**
+         * @param clientSecret
+         */
+        public TokenSignatureValidationBuilder clientsecret(@Sensitive String clientSecret) {
+            this.clientsecret = clientSecret;
+            return this;
+        }
+
+        /**
+         * @param issuer
+         * @return
+         */
+        public TokenSignatureValidationBuilder issuer(String issuer) {
+            this.issuer = issuer;
+            return this;
+        }
+
+        /**
+         * @return
+         * @throws Exception
+         * @throws InterruptedException
+         * @throws IOException
+         * @throws PrivilegedActionException
+         * @throws KeyStoreException
+         */
+        public Key getVerificationKey() throws KeyStoreException, PrivilegedActionException, IOException, InterruptedException, Exception {
+            String kid = this.signature.getKeyIdHeaderValue();
+            String x5t = this.signature.getX509CertSha1ThumbprintHeaderValue();
```

Comment:
In another PR, perhaps reading the kid and x5t values could be done after checking for "HS" algorithm since they won't be needed if the algorithm is "HS" based.